src/ATen/native/xpu/sycl/ActivationGluKernels.cpp

```diff
@@ -0,0 +1,75 @@
+#include <ATen/ATen.h>
+
+namespace at::native::xpu {
+
+template <typename scalar_t>
+static void GatedLinearUnit_updateOutput(
+    Tensor& output,
+    const Tensor& input,
+    int64_t dim) {
+  auto wrap_dim = maybe_wrap_dim(dim, input.dim());
+  const int64_t nln = input.size(wrap_dim);
+  const int64_t inputSize = nln / 2;
+  Tensor firstHalf = input.narrow(wrap_dim, 0, inputSize);
+  Tensor secondHalf = input.narrow(wrap_dim, inputSize, inputSize);
+  // output = output + firstHalf * sigmoid(secondHalf)
+  Tensor sigNum = at::empty_like(secondHalf);
+  at::sigmoid_out(sigNum, secondHalf);
+  output = at::mul(firstHalf, sigNum);
+}
+
+void glu_kernel(const Tensor& self, int64_t dim, Tensor& out) {
+  AT_DISPATCH_FLOATING_TYPES_AND2(
+      at::ScalarType::BFloat16,
+      at::ScalarType::Half,
+      self.scalar_type(),
+      "glu_xpu",
+      [&] { GatedLinearUnit_updateOutput<scalar_t>(out, self, dim); });
```

Comment:
We cannot follow IPEX implementation. Since it brings a lot of kernels. Please align with CUDA implementation.
<img width="577" alt="image" src="https://github.com/intel/torch-xpu-ops/assets/28250760/1dd5df02-6c88-4397-93a8-c2264d7aa72a">